Bound index to number of children.

MatejJan opened the pull request:
Because of imprecisions in calculating the index, it would sometimes be
out of bounds on the upper end too, so this max/min combination
constrains it between 1 and number of children.

MatejJan committed:
Bound index to number of children.

Because of imprecisions in calculating the index, it would sometimes be
out of bounds on the upper end too, so this max/min combination
constrains it between 1 and number of children.

MatejJan commented:
Let me know if you prefer to have two if statements instead of the one-liner.

```
index = 1 if index < 1
index = numberOfChildren if index > numberOfChildren
```

MatejJan committed:
Vertical scrubbing direction added.

There is a combined option as well, which chooses from horizontal or
vertical direction, depending on the side from which you move the mouse
in.

MatejJan commented:
I've added another commit that adds support for vertical scrubbing direction or a combination of horizontal/vertical (depending on which side you hover from with the mouse). Feel free to use the code if you find this useful or cherry pick just the first commit to get the bug fixed.

MatejJan committed:
Direction selection bug fix.

marco-martins commented:
Hi Matej, thanks for your collaboration. 
I will do the merge, give me a bit more time to dig into that. ;)

MatejJan committed:
Better combined direction handling.

Now the active direction can switch depending on the current direction
you’re moving.

MatejJan commented:
I submitted some more code that improves how combined direction works and now reacts to user changing direction while hovering over the scrubber (before it only mattered where they entered from).

I also unified the comment style to use ### everywhere. Again, feel free to cherry pick if this is useful to you.

MatejJan committed:
More compact code.
